libvast/src/type.cpp

```diff
@@ -3305,7 +3305,12 @@ int sum_type_access<arrow::DataType>::index_from_type(
                               caf::detail::tl_size<extension_types>::value>());
   // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
   auto result = table[x.id()];
-  VAST_ASSERT(result != unknown_id, "unexpected Arrow type id");
+  VAST_ASSERT(result != unknown_id, fmt::format("unexpected Arrow type id '{}' "
```

Comment:
since `x` is of type `arrow::DataType&` I'm just doing `x.ToString()` three lines down. What additional information could we print here based on visiting the id?